pkg/storage/replica_state.go

```diff
@@ -210,6 +210,47 @@ func setAppliedIndex(
 		nil /* txn */)
 }
 
+func setAppliedIndexBlind(
```

Comment:
Add comments to these methods explaining what's going on. Do we still need the non-blind version?